sanitycheck: error on duplicate board identifiers

nashif opened the pull request:
2 boards had duplicate identifiers, fix those and add a check in sanitycheck to error when such issues are found.

nashif committed:
boards: ODROID-GO: fix board identifier

Using the esp32 identifier which is wrong. Probably a copy/paste error.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: LPCXpresso55S69: remove duplicate identifier

Use unique identifier for this configuration. cpu0 is already used.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: error on duplicate board identifier

Error when we have duplicate identifiers in board definition files.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

chrta commented on scripts/sanity_chk/sanitylib.py:
```diff
@@ -2564,6 +2564,9 @@ def add_configurations(self):
                     platform.load(file)
+                    if platform.name in [p.name for p in self.platforms]:
+                        logger.error(f"Duplicate platform {platform.name} in {file}")
+                        #raise Exception(f"Duplicate platform identifier {platform.name} found")
```
Is this comment intended?

nashif replied:
fixed..

galak commented:
LGTM, same question as @chrta regarding the commented out code.

jenmwms commented:
I don't know if it is a blocking issue, but there are some `samples/` are getting cmake build error for the `lpcxpresso55s69_cpu1`, which I can reproduce locally. It works for `lpcxpresso55s69_cpu0`. Is that a problem?

nashif commented:
> I don't know if it is a blocking issue, but there are some `samples/` are getting cmake build error for the `lpcxpresso55s69_cpu1`, which I can reproduce locally. It works for `lpcxpresso55s69_cpu0`. Is that a problem?

@jenmwms  yes, can you please open a bug an assign it to @MaureenHelm ? :-)

jenmwms commented:
> @jenmwms yes, can you please open a bug an assign it to @MaureenHelm ? :-)

Thanks for suggesting! Please see https://github.com/zephyrproject-rtos/zephyr/issues/27880 capturing this.

nashif committed:
boards: lpcxpresso55s69_cpu1: disable sanitycheck

This board is broken and is preventing us from fixing issues that
allowed it to be merged skipping CI checks. Disabling the board from
sanitycheck to avoid more boards with the same issue from happening
again.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
lpcxpresso55s69_cpu1 needs some work still and it is preventing this PR from going in. Disabled sanitycheck for that board now (it never built in sanitycheck before anyways) until it is fixed but allowing the other fixes to go in so we catch issues that cause this to happen.

abrodkin commented:
nSIM part is already addressed here https://github.com/zephyrproject-rtos/zephyr/pull/28379, so should be dropped from this one.

abrodkin reviewed (changes requested):
Let's get rid of nsim part, as it's addressed here https://github.com/zephyrproject-rtos/zephyr/pull/28379 separately.

nashif commented:
> Let's get rid of nsim part, as it's addressed here #28379 separately.

then you need to go approve that one.. :-)

abrodkin commented:
@nashif done!

nashif commented:
> @nashif done!

ok, patch removed and other PR merged.
